base/int.jl

```diff
@@ -339,10 +339,10 @@ typemax(::Type{UInt64}) = 0xffffffffffffffff
 @eval typemin(::Type{Int128} ) = $(convert(Int128,1)<<127)
 @eval typemax(::Type{Int128} ) = $(box(Int128,unbox(UInt128,typemax(UInt128)>>1)))
 
-widen{T<:Union{Int8, Int16}}(::Type{T}) = Int
+widen{T<:Union{Int8, Int16}}(::Type{T}) = Int32
```

Comment:
Yes, that would have been my first choice as well. However, (a) current behaviour widens to larger integer types, and (b) 16-bit operations are sometimes slower than 32-bit operations (on modern Intel CPUs).

On the other hand, if we go for auto-vectorization, then keeping the type as small as possible would be beneficial.